client-hc/src/main/java/com/graphhopper/api/GHMatrixAbstractRequester.java

```diff
@@ -146,24 +153,36 @@ protected void fillResponseFromJson(MatrixResponse matrixResponse, JsonNode solu
             }
 
             JsonNode distancesFromArray = null;
-            int[] distances = null;
+            double[] distances = null;
```

Comment:
this was cast to int for a moment and later its a double again so I changed this.

Reply:
But why would we need double here (8 byte) when we store distances in meter (0 to 10 000 000)?

Reply:
Yes, as we discussed before it would be best to store distances as ints in units of 0.1m. In this case precision should be sufficient also for indoor stuff and the maximum distance should be enough also for very long shortcut edges. Here I changed this to double because the numbers stored were retrieved as floating point numbers (from json) and also `MatrixResponse#getDistance()` returns a double. Whatever you like better I can change this back to int as well. In the long run we should switch all distances to ints in my opinion.

Reply:
Yeah, I would keep this as int[] here. 

IMO we should differentiate between storage and API. The Java API should stay as `double` and only the storage should be int (or less). For edge.getDistance it could be valuable to investigate if we can include it in IntsRef. (btw: there is an interesting difference to e.g. max_speed or road_class then: the distance needs to be splitted for the virtual edges in the QueryGraph. And that is also the reason I did not try this yet ;) )

Reply:
Ok I changed back to int[]. Also opened #1663.